### ISSUE TITLE
What's the benefit over $_SESSION ?

### ISSUE THREAD

**Issue by faresite** [ericmann/wp-session-manager#38]
Hello there,
Might be a very stupid question, so I apologise, but how is this better than $_SESSION itself?
Thanks!

**Comment by ericmann** [comment]
It fits in a very specific niche for load balanced sites. If you have session data you want to store on a per-user basis, but you're running multiple servers behind a load balancer, `$_SESSION` could be problematic. By default, PHP writes its session data to _disk_, meaning your different servers have different session stores entirely.

If you use sticky sessions, this is a non-issue because you can make the load balancer route visitors back to the server they hit first (meaning they always get their session back). In other situations, tho, (i.e. round-robin load balancing) there's no guarantee a visitor will hit the same server when they change the page. If they do hit a new server, their session is effectively lost.

**TL;DR;** The entire point of this project was to emulate `$_SESSION` storage but use the database as the storage engine as it's common between even load-balanced servers.

**Comment by faresite** [comment]
Thank you very much for your answer! That makes a lot of sense, wasn't aware that it uses the database.
